src/Normalizer/TextNormalizer.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the league/commonmark package.
+ *
+ * (c) Colin O'Dell <colinodell@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace League\CommonMark\Normalizer;
+
+/***
+ * Normalize text input using the steps given by the CommonMark spec to normalize labels
+ *
+ * @see https://spec.commonmark.org/0.29/#matches
+ */
+final class TextNormalizer implements TextNormalizerInterface
+{
+    /**
+     * @var array<int, array<int, string>>
+     *
+     * Source: https://github.com/symfony/polyfill-mbstring/blob/master/Mbstring.php
+     */
+    private const CASE_FOLD = [
+        ['µ', 'ſ', "\xCD\x85", 'ς', "\xCF\x90", "\xCF\x91", "\xCF\x95", "\xCF\x96", "\xCF\xB0", "\xCF\xB1", "\xCF\xB5", "\xE1\xBA\x9B", "\xE1\xBE\xBE", "\xC3\x9F", "\xE1\xBA\x9E"],
+        ['μ', 's', 'ι',        'σ', 'β',        'θ',        'φ',        'π',        'κ',        'ρ',        'ε',        "\xE1\xB9\xA1", 'ι',            'ss',       'ss'],
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize(string $text, $context = null): string
+    {
+        // Collapse internal whitespace to single space and remove
+        // leading/trailing whitespace
+        $text = \preg_replace('/\s+/', ' ', \trim($text));
+
+        if (!\defined('MB_CASE_FOLD')) {
+            // We're not on a version of PHP (7.3+) which has this feature
```

Comment:
We should really have the same behaviour on all php versions? Maybe some of the code to do this can be copied over from https://github.com/symfony/polyfill/blob/master/src/Mbstring/Mbstring.php#L272?

Reply:
Oh, this is literally what you have done! Ha, I should go to sleep. ;)

Reply:
Hehe :upside_down_face: 

The only inconsistency was that we were converting things to upper-case, whereas PHP 7.3+ seems to use lower-case, so I've made that change.  But everything else matches the polyfill.

There are still some edge cases [that even Symfony's polyfill doesn't handle](https://github.com/symfony/polyfill/issues/173) but it should be close enough.  I don't think it's enough to warrant a bump in the minimum PHP version in 2.x (we're going from 7.1 to 7.2 as of now) as this logic here is "good enough" for those users.

Reply:
Yeh. :)